src/main/java/io/vertx/core/http/StreamPriority.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2011-2018 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.http;
+
+import io.netty.handler.codec.http2.Http2CodecUtil;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * This class represents HTTP/2 stream priority defined in RFC 7540 clause 5.3
+ */
+@DataObject
+public class StreamPriority {
+    
+    public static final StreamPriority DEFAULT = new StreamPriority();
+    /**
+     * 
+     */
+    private short weight;
+    private int streamDependency;
+    private boolean exclusive;
+    
+    private StreamPriority() {
+        this(0, Http2CodecUtil.DEFAULT_PRIORITY_WEIGHT, false);
+    }
+    
+    public StreamPriority(JsonObject json) {
+        this.weight = json.getInteger("weight", (int)Http2CodecUtil.DEFAULT_PRIORITY_WEIGHT).shortValue();
```

Comment:
generated instead of converter (with code generation) so naming follows the converter convention.

Reply:
This one is a bit problematic. I intentionally have not added setters to StreamPriority, so you can't do something like:
req.getStreamPriority().setWeight(4)
req.getStreamPriority().setStreamDependecy(5)
If you do this the priority frame would not be sent which would be confusing.

If I do not have setters or setter access is not public the generator generates converter like this:
   static void fromJson(Iterable<java.util.Map.Entry<String, Object>> json, StreamPriority obj) {
    for (java.util.Map.Entry<String, Object> member : json) {
      switch (member.getKey()) {
        case "exclusive":
          break;
        case "streamDependency":
          break;
        case "weight":
          break;
      }
    }
  }

Therefore I think it would be better to not use the generator or update it so it works with package accessible methods.